mantidproject/mantid
# New algorithm: SaveGDA

A new algorithm has been implemented to save a focused diffraction workspace to the MAUD-readable GDA format.

This algorithm has a few quirks to it, that could be with being explained (they're also cleared up in the documentation). The big one is that GDA is a TOF format, but the TOF values that SaveGDA writes out are not necessarily the actual recorded ones.

This is because of the following. MAUD reads in a TOF file and converts it to D using a calibration file, similar to a GSAS calibration file (ie containing DIFC conversion factors). The use case for this is a 160-bank workspace. We can't save the actual TOF output, as it is not practical to produce a calibration file for this many banks - there just isn't enough data in each of the banks for such runs. Instead, we use the 6-bank calibration file and essentially manually undo AlignDetectors. Since conversion is done with the same parameters when going from D to TOF (in Mantid) as when going back (in MAUD), the end result is still correct.

After this algorithm is merged, the final step will be to create the MAUD calibration file as part of the texture reduction - it will be at this point that we'll be able to make sure that the same conversion factors are used for both conversions.

**To test:**

The GDA format has been validated by scientists, so code review and a quick crack at breaking it should be fine

Test data is 
[PR22578.zip](https://github.com/mantidproject/mantid/files/2102017/PR22578.zip)

Make sure the following runs OK

```
input_ws = Load(Filename="GEM61785_focused_texture_d.nxs")
SaveGDA(InputWorkspace=input_ws,
        GSASParamFile="GEM_PF1_PROFILE.IPF",
        GroupingScheme=[1] * 3 + [2] * 8 + [3] * 20 + [4] * 42 + [5] * 52 + [6] * 35,
        Filename="/somewhere.gda")
```

Closes #22468

Does this update require release notes?
[Yes](https://github.com/mantidproject/mantid/pull/22578/commits/3f8e311de4ec94a099655d93baacde699f3e66c0)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
